diff -uNrp ./x264.004/input/y4m.c ./x264.005/input/y4m.c
--- ./x264.004/input/y4m.c	2018-05-03 14:47:09.934083700 +0300
+++ ./x264.005/input/y4m.c	2018-05-05 00:16:57.547313800 +0300
@@ -75,10 +75,12 @@ static int open_file( char *psz_filename
     int colorspace = X264_CSP_NONE;
     int alt_colorspace = X264_CSP_NONE;
     int alt_bit_depth  = 8;
+    int found_length = 0;
     if( !h )
         return -1;
 
     info->vfr = 0;
+    info->num_frames = 0;
 
     if( !strcmp( psz_filename, "-" ) )
         h->fh = stdin;
@@ -170,6 +172,14 @@ static int open_file( char *psz_filename
                     tokstart += 6;
                     alt_colorspace = parse_csp_and_depth( tokstart, &alt_bit_depth );
                 }
+                if( !strncmp( "LENGTH=", tokstart, 7 ) )
+                {
+                    tokstart += 7;
+                    if( sscanf( tokstart, "%d", &info->num_frames ) == 1 ) {
+                        x264_cli_log( "y4m", X264_LOG_DEBUG, "found header LENGTH: %d.\n", info->num_frames );
+                        found_length = 1;
+                    }
+                }
                 tokstart = strchr( tokstart, 0x20 );
                 break;
         }
@@ -192,7 +202,6 @@ static int open_file( char *psz_filename
     FAIL_IF_ERROR( h->bit_depth < 8 || h->bit_depth > 16, "unsupported bit depth `%d'\n", h->bit_depth );
 
     info->thread_safe = 1;
-    info->num_frames  = 0;
     info->csp         = colorspace;
 
     if( h->bit_depth > 8 )
@@ -223,7 +232,9 @@ static int open_file( char *psz_filename
         fseek( h->fh, 0, SEEK_END );
         uint64_t i_size = ftell( h->fh );
         fseek( h->fh, init_pos, SEEK_SET );
-        info->num_frames = (i_size - h->seq_header_len) / h->frame_size;
+        /* Don't need to check this if header LENGTH is present */
+        if( !found_length )
+            info->num_frames = (i_size - h->seq_header_len) / h->frame_size;
         FAIL_IF_ERROR( !info->num_frames, "empty input file\n" );
 
         /* Attempt to use memory-mapped input frames if possible */
